Realm/RLMProperty.h

```diff
@@ -92,4 +97,17 @@ RLM_ASSUME_NONNULL_BEGIN
 
 @end
 
+
+/**
+ This class describes a specific property on a given class.
+ */
+@interface RLMPropertyDescriptor : NSObject
+
++ (instancetype)descriptorWithClass:(Class)objectClass propertyName:(NSString *)propertyName;
```

Comment:
There was a recently-accepted proposal to fix that, but Swift 3 will also have `#keypath(Person.dogs)` for a checked keypath string.